.meta/sinks/loki.toml

```diff
@@ -0,0 +1,133 @@
+[sinks.loki]
+title = "loki"
+beta = true
+common = false
+delivery_guarantee = "best_effort"
+egress_method = "batching"
+function_category = "transmit"
+healthcheck = true
+input_types = ["log"]
+write_to_description = "[Loki][urls.loki]"
+
+<%= render("_partials/_component_options.toml", type: "sink", name: "loki") %>
+
+<%= render("_partials/_batch_options.toml", namespace: "sinks.loki.options", common: false, max_events: nil, max_size: 10490000, timeout_secs: 1) %>
+
+<%= render("_partials/_buffer_options.toml", namespace: "sinks.loki.options") %>
+
+<%= render(
+  "_partials/_request_options.toml",
+  namespace: "sinks.loki.options",
+  common: false,
+  in_flight_limit: 5,
+  rate_limit_duration_secs: 1,
+  rate_limit_num: 5,
+  retry_attempts: -1,
+  retry_initial_backoff_secs: 1,
+  retry_max_duration_secs: 10,
+  timeout_secs: 60
+) %>
+
+<%= render("_partials/_tls_options.toml", namespace: "sinks.loki.options", can_enable: false, can_verify: true) %>
+
+[sinks.loki.options.host]
+type = "string"
+required = true
+examples = ["http://localhost:3100", "http://127.0.0.1:8080"]
+description = "The host used to ship logs to."
+
+[sinks.loki.options.encoding]
+type = "string"
+common = true
+required = true
+description = """\
+The encoding format used to serialize the events before outputting.\
+"""
+
+[sinks.loki.options.encoding.enum]
+json = "Each event is encoded into JSON"
+text = "Each event is encoded into text via the `message` key."
+
+[sinks.loki.options.tenant_id]
+type = "string"
+required = false
+examples = ["some_tenant_id"]
+description = """\
+The tenant id that will be sent with every request, by default this is not required since a\
+proxy should set this header. When running loki locally a tenant id is not required either.\
+
+You can read more about tenant id's [here][urls.loki_multi_tenancy]\
+"""
+
+[sinks.loki.options.labels]
+type = "table"
+required = true
+examples = [{label = "value"}]
+description = """\
+A set of labels that will be attached to each batch of events. These values\
```

Comment:
Maybe we should remind users to avoid to use high cardinality labels like they would in Prometheus. I mean may be this is obvious.

Do you guys have some sort of label/tag discovery from k8s ?

Reply:
We're about to. @ktff is working on a `kubernetes_metadata` transform that should be completed in the next week or 2, and we currently have [these enrichment transforms](https://vector.dev/components/?functions[]=enrich).

Reply:
Cool this should make it to Loki sink !

Reply:
:+1: I will add a note about high cardinality labels.

So we don't really have a concept of a schema for Kubernetes labels but we are about to provide a transform that can enrich your events with kube labels. It may make sense to add this to our docs to explain how you can combine that transform with this sink.

The idea is that you can configure loki like so:

```toml
[sinks.my_loki]
# ~~ snip other config items ~~
labels = {pod_id = "{{ labels.pod_id }}"}
```

Related k8 issue https://github.com/timberio/vector/issues/1072

cc @binarylogic this is kinda in the realm of guide vs sink doc page, I think it makes sense to add a note about kube possibly within the loki sink docs? What do you think?

Reply:
Yep, it is. I've got it and thanks for the heads up.